adjtimex.c:18:15: error: absolute value function 'abs' given an argument of type 'long' but has parameter of type 'int' which may cause truncation of value [-Werror,-Wabsolute-value]

I just tried to build rr with yesterday's clang trunk (clang version 5.0.0 from svn revision 304217).

The build failed at this build warning treated as an error:
````
/scratch/work/tools/rr/src/test/adjtimex.c:18:15: error: absolute value function 'abs' given an argument of type 'long' but has parameter of type 'int' which may cause truncation of value [-Werror,-Wabsolute-value]
  test_assert(abs(tx.time.tv_sec - ts.tv_sec) <= 1);
              ^
/scratch/work/tools/rr/rr/src/test/adjtimex.c:18:15: note: use function 'labs' instead
  test_assert(abs(tx.time.tv_sec - ts.tv_sec) <= 1);
              ^~~
              labs
/scratch/work/tools/rr/rr/src/test/rrutil.h:163:60: note: expanded from macro 'test_assert'
#define test_assert(cond) assert("FAILED: !" && check_cond(cond))
                                                           ^
/usr/include/assert.h:89:5: note: expanded from macro 'assert'
  ((expr)                                                               \
    ^
1 error generated.
````

--- a/src/test/adjtimex.c
+++ b/src/test/adjtimex.c
@@ -15,7 +15,7 @@ int main(void) {
   test_assert(0 == clock_gettime(CLOCK_REALTIME, &ts));
 
   // Verify that adjtimex() and clock_gettime() return roughly the same time.
-  test_assert(abs(tx.time.tv_sec - ts.tv_sec) <= 1);
+  test_assert(labs(tx.time.tv_sec - ts.tv_sec) <= 1);
 
   atomic_puts("EXIT-SUCCESS");
   return 0;

